packages/kbn-dev-utils/src/kbn_node/get_script_source.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { platform } from 'os';
+import { REPO_ROOT } from '../repo_root';
+
+const SCRIPT_SOURCE_SH = `#!/bin/sh
+#
+# ** THIS IS AN AUTO-GENERATED FILE **
+# ** PLEASE DO NOT CHANGE IT MANUALLY **
+#
+# GENERATED BY KIBANA \`node scripts/register_kbn_node\`
+# IF YOU WANNA CHANGE SOMETHING IN THIS SCRIPT
+# PLEASE RE-RUN 'yarn kbn bootstrap' or 'node scripts/register_kbn_node'
+
+DIR=${REPO_ROOT}
+CONFIG_DIR=\${KIBANA_PATH_CONF:-"$DIR/config"}
+
+# Setup node dev options file to node.dev.options
+NODE_DEV_OPTIONS_FILE=\${CONFIG_DIR}/node.dev.options
+
+# Fallback to node.ci.options in case node.dev.options not exists
+if [ ! -f "$NODE_DEV_OPTIONS_FILE" ]; then
+  NODE_DEV_OPTIONS_FILE=\${CONFIG_DIR}/node.ci.options
+fi
+
+NODE_OPTIONS="$(grep -v ^# < $NODE_DEV_OPTIONS_FILE | xargs) $NODE_OPTIONS" exec "node" \${@}
```

Comment:
this may not be the PR for it, but i'd like to propose renaming `NODE_OPTIONS` to `KBN_NODE_OPTS`.  1) consistency with elasticsearch and 2) avoid global conflicts

no issues with postponing - i just had the thought.

Reply:
I might be missing something but could you give a more in depth example of what you want to do? The final exported/set variable needs to be `NODE_OPTIONS` otherwise it won't be picked by node https://nodejs.org/api/cli.html#cli_node_options_options

Reply:
the comparable version to elasticsearch would be something like `node $KBN_NODE_OPTS src/cli`, [ref](https://github.com/elastic/elasticsearch/blob/3950c3a9da3a2806d7a31db3c964960cd58961d1/docs/reference/setup/jvm-options.asciidoc).

i'm not sure we want to go that far here, that's a breaking change.  but maybe `NODE_OPTIONS=$NODE_OPTIONS $KBN_NODE_OPTIONS`, with any exports and documentation referring to the kbn version.  just a namespace for setting kibana only options that won't leak

Reply:
I would say we should left that possible change for other PR maybe for 8.0 only in order to not introduce any breaking change.